packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart

```diff
@@ -993,6 +996,46 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface
         .result;
   }
 
+  @override
+  Future<CognitoDevice> fetchCurrentDevice() async {
+    final tokens = await stateMachine.getUserPoolTokens();
+    final deviceSecrets = await _deviceRepo.get(tokens.username);
+    final deviceKey = deviceSecrets?.deviceKey;
+    if (deviceSecrets == null || deviceKey == null) {
+      throw const DeviceNotTrackedException();
+    }
+
+    late GetDeviceResponse resp;
+
+    try {
+      resp = await _cognitoIdp
+          .getDevice(
+            cognito.GetDeviceRequest(
+              deviceKey: deviceKey,
+              accessToken: tokens.accessToken.raw,
+            ),
+          )
+          .result;
+    } on Exception catch (error) {
+      throw AuthException.fromException(error);
+    }
+
+    final device = resp.device;
+    final attributes =
+        device.deviceAttributes ?? const <cognito.AttributeType>[];
+
+    return CognitoDevice(
+      id: deviceKey,
+      attributes: {
+        for (final attribute in attributes)
+          attribute.name: attribute.value ?? '',
+      },
```

Comment:
Q: Does this need a for loop? Does `attributes` have other properties? 
```suggestion
      attributes: attributes,
```

Reply:
> Q: Does this need a for loop? 

No, you can rewrite this (after checking attributes is [not empty](https://pub.dev/documentation/built_collection/latest/built_collection/BuiltList/isNotEmpty.html) like this:
```
    final test = attributes.elementAt(0);
    final name = test.name;
    final value = test.value ?? '';
    ...
    attributes = {name: value}
```
... among other methods in the [BuiltList class](https://pub.dev/documentation/built_collection/latest/built_collection/BuiltList-class.html). The loop is mostly stylistic and for parity, being that this is how the other devices API's extracts the name and values from the attributes map. 

Note that `attributes: attributes` is type invalid.  

> Does `attributes` have other properties?

As of now, no

Reply:
Makes sense, thanks for adding that context.